### ISSUE TITLE
  Hi How do I convert lists of C. elega...

### ISSUE THREAD

**Issue by wormbasehelp** [WormBase/website#4398]
*Submitted from the feedback form on the WormBase website.*

<p>&nbsp;&nbsp;Hi<br/>How do I convert lists of C. elegans gene IDs  (i.e.: H23N18.1,  C08E7.9, T12B5.10, C52A10.1)  into NCBI GeneID??<br/>This is so I can submit my lists of DE genes from RNA-Seq output to online KEGG for pathway enrichment analysis.<br/>Many thanks</p><p>What were you doing? <br />&nbsp;&nbsp;This is so I can submit my lists of DE genes from RNA-Seq output to online KEGG for pathway enrichment analysis.</p>

**Reported by:** Susa********* (stas****************)
**Submitted from:** <a target="_blank" href="http://www.wormbase.org//tools/support">/tools/support</a>
**Browser:** Mozilla/5.0 (Windows NT 6.1; WOW64; rv:40.0) Gecko/20100101 Firefox/40.0

**Comment by chris-grove** [comment]
@brucebolt  This help desk question made me realize that the ParaSite BioMart does not recognize "sequence names" for genes that already have a public gene name.
If you use the BioMart Filters' Gene section "ID list limit", from the drop down menu you can select "Gene stable ID(s)", "Gene Name(s)", "Transcript stable ID(s)", and "Protein stable ID(s)"
The problem is that the BioMart will not recognize the gene name "H23N18.1" as a gene name, even though this is a valid sequence name for the gene with a public name "ugt-13". "H23N18.1" can be entered and recognized if I use the "Transcript stable ID(s)" drop down option, but this only works because this gene only has one transcript, and so the one and only transcript has the same name as the gene's sequence name. This will not work for any gene sequence names for genes that have multiple transcripts.

Bottom line: BioMart needs to recognize both the public name, e.g. "ugt-13" AND the sequence name, e.g. "H23N18.1" as a valid gene name for the same gene

**Comment by chris-grove** [comment]
Hello Susan,

 This can be done using the WormBase ParaSite BioMart tool:

http://parasite.wormbase.org/biomart/martview/f6d15c79bdc91f4db9f84b729c26e703

reachable from the "Tools" menu on the WormBase homepage. Unfortunately,
it looks like BioMart has a problem recognizing sequence names of genes
that also have a public name (e.g. H23N18.1 versus ugt-13). We are working
to fix that.

In the meantime, could you send me your list of genes and we can convert
the list for you?

I'm happy to show you how to perform the query using BioMart.

Best,

Chris Grove
WormBase

**Comment by chris-grove** [comment]
User responded and I've given her instructions on how to perform the query in BioMart

**Comment by brucebolt** [comment]
I have added WormBase Sequence Name and also Locus ID as both filters and attributes to the ParaSite mart.  This is updated on the live site.

**Comment by epaule** [comment]
From past experiences, restricting the species in the filter to C.elegans
makes a big difference for BioMart query times.

M

**Comment by khowe** [comment]
Yes, restricting to C. elegans makes all the difference with this query. 

However, when you give an id list, you shouldn't need to restrict on species. Can this be solved by adding further indices?

**Comment by brucebolt** [comment]
For some unknown reason, a number of indices in the underlying MySQL database have been dropped since production.  These are being recreated now.

I'll take this opportunity to see if there are any other logical columns which could be indexed, in order to speed up performance across all queries.

**Comment by khowe** [comment]
Hmm. Is this related to the earlier problem we observed with the compara database, where certain tables became unoptimized after mirroring to live?

**Comment by brucebolt** [comment]
This is unrelated - mysqlnaga is not used on the mart database.

What is puzzling me is that only some tables have lost their indices.  And in some cases, only some indices have been lost.  The loss appeared to happen in the Hinxton datacentre, which was subsequently mirrored to London (live) when I made the template update yesterday.

**Comment by brucebolt** [comment]
Indices have been recreated and synched to the live site.  The performance has vastly improved and searching by WB Sequence Name now completes quickly.

**Comment by chris-grove** [comment]
OK, working well. Thanks @brucebolt !
